src/robots/yamcs_TMTC.py

```diff
@@ -82,56 +95,22 @@ def _drive_robot_turn(self, angular_velocity, angle):
         self._stop_robot_after_time(turn_time * turn_time_adjustment_coef)
 
     def _stop_robot_after_time(self, travel_time):
-        start_time = self.timeline.get_current_time()
-        self._stop_time = start_time + travel_time
-        
-        if self._drive_callback_sub is None:
-            # in case of receiving a command before the previous one was completed, the subscribed callback
-            # will remain, and will use the updated _stop_time
-            self._drive_callback_sub = self._update_stream.create_subscription_to_pop(
-                self._stop_robot_callback,
-                name="RobotDriveStopCallback", 
-            )
-
-    def _stop_robot_callback(self, e):
-        # callback function for stopping the robot
-        # checks every frame if simulaiton has reached the self._stop_time (calculated in the caller function)
-        if self._stop_time is None:
-            return
-
-        current_time = self.timeline.get_current_time()
-
-        if current_time >= self._stop_time:
-            self._stop_robot()
+        if self._intervals_handler.does_exist(IntervalName.STOP_ROBOT.value):
+            self._intervals_handler.update_next_time(IntervalName.STOP_ROBOT.value, travel_time)
+        else:
+            self._intervals_handler.add_new_interval(IntervalName.STOP_ROBOT.value, travel_time, False, False,
+                                                 self._stop_robot)
 
     def _stop_robot(self):
         self._robot.stop_drive()
-        self._stop_time = None
-
-        if self._drive_callback_sub is not None:
-            self._drive_callback_sub.unsubscribe()
-            self._drive_callback_sub = None
-
-    def start(self):
-        # initially inteded to be in a for robot in robots loop, thus to have one thread for each robot
-        # however, for the workshop use-case, the code was simplified to assume use of only one robot 
-        t = threading.Thread(
-            target=self._yamcs_transmitter,
-            args=(self._robot_name, self._yamcs_conf["interval_s"]),
-            name="yamcs-TMTC-" + self._robot_name,
-            daemon=True,
-        )   
-        t.start()
-
-    def _yamcs_transmitter(self, robot_name, interval_s):
-        print("started TMTC for: " + robot_name)
-        try:
-            while True:
-                self._transmit_pose_of_base_link()
-                # add here further commands
-                time.sleep(interval_s) #TODO: change into simulation secs
-        finally:
-            print("ended transmitter for: " + robot_name)
+        self._intervals_handler.remove_interval(IntervalName.STOP_ROBOT.value)
+
+    def start_streaming_data(self):
+        self._intervals_handler.add_new_interval(IntervalName.POSE_OF_BASE_LINK.value, self._yamcs_conf["intervals"]["robot_stats"], True, True,
```

Comment:
for legibility: 
the function add_new_intervals() has many arguments. and their raw value e.g True, True is not immediately obvious. in this case, call the function using keyword arguments instead of relying only on positional arguments.
e.g add_nuew_interval(
    name=...,
    seconds=...,
etc